lib/iris/mesh/components.py

```diff
@@ -3017,6 +2993,69 @@ def summary(self, *args, **kwargs):
             result = "\n".join(lines)
         return result
 
+    def _load_points_and_bounds(self):
+        points, bounds = self._construct_access_arrays()
+        if points is None:
+            # TODO: we intend to support this in future, but it will require
+            #  extra work to refactor the parent classes.
+            msg = "Cannot yet create a MeshCoord without points."
+            raise ValueError(msg)
+
+        # Get the 'coord identity' metadata from the relevant node-coordinate.
```

Comment:
Something missed in #4757 and in previous discussions: the metadata such as `units`, `standard_name`, etcetera is also subject to change.

Thankfully it is quick to recalculate, so we don't need to protect it behind a date check. But we do need a way to make sure it is recalculated on the fly.

Metadata classes a tricksy and it's probably best if we don't make any changes there. Instead we can put `self._metadata_manager` behind a `@property' so that we can regenerate it every time before returning it:


#### 1

```diff
- self._metadata_manager = metadata_manager_factory(MeshCoordMetadata)
+ self._some_other_name = metadata_manager_factory(MeshCoordMetadata)
```

#### 2

```python
@property
def _metadata_manager(self):
    # An explanatory comment.
    
    self._some_other_name.standard_name = something
    # Etcetera for all standard coord metadata
    # THIS INCLUDES DETERMINING THE CORRECT VALUE, AS IN
    #  THE CURRENT BLOCK WITHIN _load_points_and_bounds

    return self._some_other_name
```

#### 3

I don't _think_ this will be necessary since I don't _think_ there will be any calls to set `self._metadata_manager`?

```python
@_metadata_manager.setter
def _metadata_manager(self, value):
    self._some_other_name = value
```